### PR TITLE
Add Python doc for Trellis

### PR DESCRIPTION
**Pull request by swalkinshaw**

### CODE REVIEW

**Comment by swalkinshaw** [comment]
Oh wait, this needs to consistent + cross-referenced in https://roots.io/docs/getting-started/macos/#requirements

**Comment by swalkinshaw** [comment]
Ok updated everything:

1. added new Python page that consolidates everything about Python, pip, ansible, trellis-cli 
2. updated Getting started pages to be consistent and link to new page

**Review by retlehs — APPROVED** [review]

File: docs/trellis/master/python.md [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+```
+
+### Ubuntu
+
+Ubuntu 20.04 comes default with Python 3 available as `python3`
+only. There's no "unversioned" `python`. If you want to install Python 2, it
+will be installed as `python2`.
+
+The [`python-is-python3`])(https://packages.ubuntu.com/focal/python-is-python3) package
+exists solely as an easy to symlinks `/usr/bin/python` to `python3`.
```
Comment (alwaysblank) [inline docs/trellis/master/python.md]:
```suggestion
The [`python-is-python3`])(https://packages.ubuntu.com/focal/python-is-python3) package
exists solely as an easy way to symlink `/usr/bin/python` to `python3`.
```

File: docs/trellis/master/python.md [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+
+### Windows
+
+Roots [recommends using WSL](https://roots.io/docs/getting-started/windows/)
+when using Windows. Since WSL uses Ubuntu, the [section](#ubuntu) above applies
+but you might need to also install `python3` (from within WSL):
+
+```bash
+sudo apt-get install -y python3 python-is-python3 python3-pip
+```
```
Comment (alwaysblank) [inline docs/trellis/master/python.md]: This is identical to the command above so "but you might need to also install `python3` (from within WSL)" is confusing (both commands install python3).

**Review by alwaysblank — CHANGES_REQUESTED** [review]
Just a minor tweak and a question. Looks good!

File: docs/trellis/master/python.md (hunk 2 above)
Comment (swalkinshaw) [reply docs/trellis/master/python.md]: Ah I updated it and forgot about this. Removed, thanks

File: docs/trellis/master/python.md [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+that the versions are only updated when macOS itself has a new major version.
+
+If you want to have more control over Python versions, we recommend using a tool
+like [pyenv](https://github.com/pyenv/pyenv) or [asdf](https://github.com/danhper/asdf-python)
+to install specific versions globally (or even per project if that's needed).
+
+We **do not recommend** installing Python from Homebrew. This might be
+surprising since it goes against most guides and recommendations but we believe
+using Python from Homebrew will cause more problems long-term due to its newer 
+"feature" of auto-upgrading packaages as described in [this article](https://justinmayer.com/posts/homebrew-python-is-not-for-you/).
```
Comment (knowler) [inline docs/trellis/master/python.md]:
```suggestion
"feature" of auto-upgrading packages as described in [this article](https://justinmayer.com/posts/homebrew-python-is-not-for-you/).
```

**Review by retlehs — CHANGES_REQUESTED** [review]
needs to be updated to account for the getting started guides being removed in favor of "getting started on [os]" instructions on the trellis installation doc (ref #360)

**Comment by swalkinshaw** [comment]
Closing since this will be folded into a new PR.

**Pull request closed without merging**